pkg/core/native/native_test/neo_test.go

```diff
@@ -771,6 +771,43 @@ func TestNEO_CalculateBonus(t *testing.T) {
 	})
 }
 
+func TestNEO_UnclaimedGas(t *testing.T) {
+	neoValidatorsInvoker := newNeoValidatorsClient(t)
+	e := neoValidatorsInvoker.Executor
+
+	acc := neoValidatorsInvoker.WithSigners(e.NewAccount(t))
+	accH := acc.Signers[0].ScriptHash()
+
+	t.Run("non-existing account", func(t *testing.T) {
+		// non-existing account, should return zero unclaimed GAS.
+		acc.Invoke(t, 0, "unclaimedGas", util.Uint160{}, 1)
+	})
+
+	t.Run("zero balance", func(t *testing.T) {
+		initialGASBalance := e.Chain.GetUtilityTokenBalance(accH)
+		h := acc.Invoke(t, true, "transfer", accH, accH, 0, nil)
+		claimTx, _ := e.GetTransaction(t, h)
+		e.CheckGASBalance(t, accH, big.NewInt(initialGASBalance.Int64()-claimTx.SystemFee-claimTx.NetworkFee))
+		acc.Invoke(t, 0, "unclaimedGas", accH, 1)
+	})
+
+	t.Run("unclaimed GAS", func(t *testing.T) {
+		amount := 100
+		defaultGASParBlock := 5
+		rewardDistance := 10
+		neoValidatorsInvoker.Invoke(t, true, "transfer", e.Validator.ScriptHash(), accH, amount, nil)
+
+		for range rewardDistance - 1 {
+			e.AddNewBlock(t)
+		}
+		expectedGas := int64(amount*rewardDistance*defaultGASParBlock) / int64(rewardDistance)
```

Comment:
It doesn't work like this. See https://github.com/nspcc-dev/neo-go/pull/3596 and https://github.com/nspcc-dev/neo-go/blob/9a383608244a822772b9e6cea25846ddd2102b92/pkg/core/native/native_neo.go#L768